monai/inferers/inferer.py

```diff
@@ -182,10 +190,16 @@ def _batch_sampler(
                 batch_size = min(self.batch_size, total_size - i)
                 yield patches[i : i + batch_size], patches[i : i + batch_size].meta[PatchKeys.LOCATION], batch_size  # type: ignore
         else:
+            buffer: Iterable | ThreadBuffer
+            if self.buffer_size > 0:
+                # Use multi-threading to sample patches with a buffer
+                buffer = ThreadBuffer(patches, buffer_size=self.buffer_size, timeout=0.01)
```

Comment:
At this point in the code is the `patches` value already a calculated list of objects? If so then this line doesn't move any computation to a separate thread. The idea with `ThreadBuffer` is that an iterable is passed as `src` which computes some result upon request. This will happen in the separate thread and so parallelises an expensive computation, however if `patches` is already computed this has already been done.

Reply:
@KumoLiu `timeout=0.01` is the default value for `ThreadBuffer`, not mentioning it also means explicitly setting it to `0.01`. We can increase it if necessary.

Reply:
@ericspod thanks for the explanation. The purpose of this ThreadBuffer here is to address loading WSI patches (`WSISlidingWindowSplitter`), which are being loaded upon calling them since the `src` here is a generator. It should not have any effect for the situations where the inputs are already in the memory (`SlidingWindowSplitter`) even if it involves computations because of GIL, right?

Reply:
The consideration here is if there's computation that can be done in parallel, so if everything is already computed and stored in memory there's no speedup. As for the GIL, this is running in a thread so will not get much speedup if the computation doesn't involve compiled code like Numpy or Pytorch which releases the GIL, and you have to consider thread-safety if data is modified in different threads. If `src` is a generator that does some computation on request we should get a speedup here yes.